#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.6822377450285608 0.6977630492321323 0.21835381009870522 0.8313115159375534
  position 1.554055860678232 1.9508838941680782 2.441748568478812
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 2 2
}
Robot {
  translation -0.070817 0.05 -0.0182776
  rotation 0 -1 0 6.53317e-06
  children [
    DEF base_link Shape {
      appearance PBRAppearance {
        baseColor 0.0659342 0.501793 0.00683604
        metalness 0
      }
      geometry Box {
        size 0.28 0.055 0.15
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 -1
        anchor 0.05 0 0.08
      }
      device [
        RotationalMotor {
          name "front_right_wheel"
        }
      ]
      endPoint Solid {
        translation 0.05 0 0.08
        rotation 0.9779513156757369 0.1476091067171582 0.14772533899862278 1.592289000087949
        children [
          DEF wheel_shape Shape {
            appearance PBRAppearance {
              baseColor 0.8 0.5 0
              metalness 0
            }
            geometry Cylinder {
              height 0.01
              radius 0.055
            }
          }
        ]
        name "front_right_wheel"
        boundingObject USE wheel_shape
        physics DEF wheel_physics Physics {
        }
        locked TRUE
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 -1
        anchor -0.08 0 0.08
      }
      device [
        RotationalMotor {
          name "rear_right_wheel"
        }
      ]
      endPoint Solid {
        translation -0.08 0 0.08
        rotation 0.9779513220677499 0.14760844352270006 0.1477259593529378 1.5922867322353302
        children [
          USE wheel_shape
        ]
        name "rear_right_wheel"
        boundingObject USE wheel_shape
        physics USE wheel_physics
        locked TRUE
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 -1
        anchor 0.05 0 -0.08
      }
      device [
        RotationalMotor {
          name "front_left_wheel"
        }
      ]
      endPoint Solid {
        translation 0.05 0 -0.08
        rotation 0.9724219821858224 0.16485048945089376 0.16498425588399962 1.5979567954753642
        children [
          USE wheel_shape
        ]
        name "front_left_wheel"
        boundingObject USE wheel_shape
        physics USE wheel_physics
        locked TRUE
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 -1
        anchor -0.08 0 -0.08
      }
      device [
        RotationalMotor {
          name "rear_left_wheel"
        }
      ]
      endPoint Solid {
        translation -0.08 0 -0.08
        rotation 0.9724221603353507 0.16485004746572918 0.16498364749052064 1.5979562588572407
        children [
          USE wheel_shape
        ]
        name "rear_left_wheel"
        boundingObject USE wheel_shape
        physics USE wheel_physics
        locked TRUE
      }
    }
    DistanceSensor {
      translation 0.14 -0.0275 0.075
      children [
        DEF ir_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "right_outer_sensor"
      boundingObject USE ir_shape
      physics DEF ir_physics Physics {
      }
      lookupTable [
        0 0 0
        0.15 1000 0
      ]
    }
    DistanceSensor {
      translation 0.14 -0.0275 0.025
      children [
        DEF ir_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "right_mid_sensor"
      boundingObject USE ir_shape
      physics DEF ir_physics Physics {
      }
      lookupTable [
        0 0 0
        0.15 1000 0
      ]
    }
    DistanceSensor {
      translation 0.14 -0.0275 -0.075
      children [
        DEF ir_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "left_outer_sensor"
      boundingObject USE ir_shape
      physics DEF ir_physics Physics {
      }
      lookupTable [
        0 0 0
        0.15 1000 0
      ]
    }
    DistanceSensor {
      translation 0.14 -0.0275 -0.025
      children [
        DEF ir_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "left_mid_sensor"
      boundingObject USE ir_shape
      physics DEF ir_physics Physics {
      }
      lookupTable [
        0 0 0
        0.15 1000 0
      ]
    }
    DistanceSensor {
      translation 0.12 -0.0275 0.075
      rotation 0 1 0 -1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "right_side_sensor"
      boundingObject USE ir_shape
      physics USE ir_physics
      lookupTable [
        0 0 0
        0.15 1000 0
      ]
    }
    DistanceSensor {
      translation 0.12 -0.0275 -0.075
      rotation 0 1 0 1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "left_side_sensor"
      boundingObject USE ir_shape
      physics USE ir_physics
      lookupTable [
        0 0 0
        0.15 1000 0
      ]
    }
    Camera {
      translation 0.14 0 0
      rotation 0 1 0 -1.57
      children [
        DEF camera_shape Shape {
          appearance PBRAppearance {
            baseColor 1 0.999969 0.999985
            metalness 0
          }
          geometry Cylinder {
            height 0.01
            radius 0.01
          }
        }
      ]
      name "front_camera"
      boundingObject USE camera_shape
      physics Physics {
      }
      locked TRUE
    }
    Solid {
      translation -0.025 0.05 -0.065
      children [
        DEF basket_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            metalness 0
          }
          geometry Box {
            size 0.2 0.1 0.01
          }
        }
      ]
      name "left basket"
      boundingObject USE basket_shape
      physics Physics {
      }
      locked TRUE
    }
    Solid {
      translation -0.025 0.05 0.065
      children [
        DEF basket_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            metalness 0
          }
          geometry Box {
            size 0.2 0.1 0.01
          }
        }
      ]
      name "right basket"
      boundingObject USE basket_shape
      physics Physics {
      }
      locked TRUE
    }
    Solid {
      translation -0.13 0.05 0
      rotation 0 1 0 1.57
      children [
        DEF basket_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            metalness 0
          }
          geometry Box {
            size 0.14 0.1 0.01
          }
        }
      ]
      name "back basket"
      boundingObject USE basket_shape
      physics Physics {
      }
      locked TRUE
    }
    Solid {
      translation 0.08 0.05 0
      rotation 0 1 0 1.57
      children [
        DEF basket_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            metalness 0
          }
          geometry Box {
            size 0.14 0.1 0.01
          }
        }
      ]
      name "front basket"
      boundingObject USE basket_shape
      physics Physics {
      }
      locked TRUE
    }
  ]
  name "GardenRobot"
  boundingObject USE base_link
  physics Physics {
  }
  controller "robot_controller"
}
